pandas/core/indexes/datetimes.py

```diff
@@ -137,11 +137,7 @@ def wrapper(self, other):
             result = func(np.asarray(other))
             result = com._values_from_object(result)
 
-            if isinstance(other, Index):
-                o_mask = other.values.view('i8') == libts.iNaT
-            else:
-                o_mask = other.view('i8') == libts.iNaT
-
+            o_mask = np.array(isna(other))
```

Comment:
why do you need the np.array?

Reply:
Test failures with older versions of numpy when o_mask is a Series.

Reply:
grr. ok